src/Plugin/Filter/TokenFilter.php

```diff
@@ -89,7 +104,23 @@ public static function create(ContainerInterface $container, array $configuratio
    * {@inheritdoc}
    */
   public function process($text, $langcode) {
-    return new FilterProcessResult($this->token->replace($text), [], ['langcode' => $langcode]);
+    $data = [];
+
+    // Attempt to figure out the current context based on the current backtrace.
```

Comment:
Ideally I can figure out the core context system in a way that it can be used for this, ideally some way of determining and setting the context when it is known in a way that is known regardless of the render method.

Worst case scenario, the original option I suggested in https://www.drupal.org/node/2075663 should still be valid in Drupal 8.

However, for the sake of moving this port forward and making bring it back to feature parity of the stable Drupal 7 release this approach should not be a blocker as it is representative of the existing approach.